### PR TITLE
chore: my collection analytics part 2 

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **Analytics**

This PR resolves [CX-923]

### Description

Adds tracking events for taps and screen views in my collection

- adds screen tracking for:  **MyCollection, MyCollectionArtwork** 
- tap tracking for:
   - **TappedSell**
   - **TappedShowMore for ‘Learn more’**
   - **TappedShowMore for ‘See all articles’**
   - **TappedShowMore for ‘Explore auction results’**
   - **TappedShowMore for ‘View more’**
 - updates cohesion to get latest schema
 - updates our analytics docs

Tap tracking for:

All these events should be implemented, just keeping a draft until I have addressed follow-ups below.

### TODO

- [x] Get clarity on if contextModules are correct
- [x] Tests on analytics firing
- [x] Update docs to advise using new cohesion screen tracking

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a [CHANGELOG.yml](/CHANGELOG.yml) entry or my changes do not require one.

[CX-923]: https://artsyproduct.atlassian.net/browse/CX-923

### CODE REVIEW

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx [hunk 1]
```diff
@@ -92,3 +103,17 @@ export const MyCollectionArtworkArtistArticlesFragmentContainer = createFragment
     `,
   }
 )
+
+const tracks = {
+  tappedShowMore: (internalID: string, slug: string, subject: string) => {
+    const tappedShowMore: TappedShowMore = {
+      action: ActionType.tappedShowMore,
+      context_module: ContextModule.myCollectionArtwork,
```
Comment (brainbicycle) [inline src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx:111]: @mikehrom is `context_module` supposed to be something more specific then this? I found some potential candidates in schema like `AuctionResults` for auction results taps but I didn't see an obvious match for all of these so I kept it the generic (basically a repeat of the screen)

Comment (mikehrom) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx:111]: Screen Shot 2020-12-22 at 2 38 49 PM

We're talking about the pink-rectangle-highlighted links for this event, right? I would be interested in defining an enumerated sub-list of context modules for my collection artworks, if we think we'll be dealing with the same ones (auction results, articles featuring, artwork details, etc.).

Something like: 
- collectedArtworkAuctionResults
- collectedArtworkDetails
- collectedArtworkArtistArticles

Comment (mikehrom) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx:111]: Alternatively, we have `subject` to provide the copy of whatever link is being tapped. That's good until the same `subject` appears in multiple contexts.

Comment (brainbicycle) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx:111]: Yep those are what I am talking about! I have the subject with copy in there but yeah "View more" is pretty generic.

These are the modules I could see existing here:
 
Here is a list of basic modules here I could see:
- relatedArticles
- header
- marketInsights
- auctionResults
- sellInfo

Comment (mikehrom) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx:111]: Where would `header` be used? The rest make sense to me (sellInfo and marketInsights would be new, right?).

Comment (mikehrom) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx:111]: I see now, we've updated from 'header' to 'artworkMetadata'

**Review by mikehrom — APPROVED** [review]

File: src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx (hunk 1 above)
Comment (brainbicycle) [reply src/lib/Scenes/MyCollection/Screens/Artwork/Components/ArtworkInsights/MyCollectionArtworkArtistArticles.tsx]: Updated with more specific module names from cohesion

**Pull request merged**